controllers/hcpvaultsecretsapp_controller.go

```diff
@@ -123,6 +139,45 @@ func (r *HCPVaultSecretsAppReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return ctrl.Result{}, r.handleDeletion(ctx, o)
 	}
 
+	shouldInitiateShadowSecretCleanupRoutine := false
+
+	orphanedShadowSecretManager.mu.Lock()
+	// since each HVPVaultSecretsApp has a unique NamespacedName, we can use the it to keep track
+	// of whether a cleanup routine is running for the HCPVaultSecretsApp already
+	if !orphanedShadowSecretManager.activeCleanupRoutines[req.NamespacedName.String()] {
```

Comment:
I didn't see a way where we were keeping track of HCPVaultSecretsApps in a list or map already. I believe that the map is necessary here because we don't want to create a new goroutine every time `Reconcile()` is executed. We only need one goroutine per HVS app to periodically check if the HVS app was deleted so that we can attempt to remove orphaned shadow secrets.